osquery/filesystem/filesystem.cpp

```diff
@@ -431,24 +432,36 @@ Status listFilesInDirectory(const fs::path& path,
 Status listDirectoriesInDirectory(const fs::path& path,
                                   std::vector<std::string>& results,
                                   bool recursive) {
-  if (path.empty() || !pathExists(path)) {
+  if (path.empty() || !pathExists(path) || !fs::is_directory(path)) {
```

Comment:
I missed this check here, so we should be moving the error code definition above this, and then pass `ec` as the second argument of `is_directory`.